calculators/ItemNetworthCalculator.js

```diff
@@ -77,168 +82,51 @@ class ItemNetworthCalculator {
-        return data;
-    }
-
-    #isRune() {
-        return (this.itemId === 'RUNE' || this.itemId === 'UNIQUE_RUNE') && this.extraAttributes.runes && Object.keys(this.extraAttributes.runes).length > 0;
-    }
-
-    #isPickonimbus() {
-        return this.itemId === 'PICKONIMBUS' && this.extraAttributes.pickonimbus_durability;
-    }
-
-    #calculatePickonimbus() {
-        const reduction = this.extraAttributes.pickonimbus_durability / pickonimbusDurability;
-        this.price += this.price * (reduction - 1);
-        this.base += this.price * (reduction - 1);
-    }
-
-    #hasHotPotatoBook() {
-        return !!this.extraAttributes.hot_potato_count;
-    }
-
-    #calculateHotPotatoBook(prices) {
-        const hotPotatoCount = Number(this.extraAttributes.hot_potato_count);
-        if (hotPotatoCount > 10) {
-            const calculationData = {
-                id: 'FUMING_POTATO_BOOK',
-                type: 'fuming_potato_book',
-                price: (prices['fuming_potato_book'] || 0) * (hotPotatoCount - 10) * applicationWorth.fumingPotatoBook,
-                count: hotPotatoCount - 10,
-            };
-            this.price += calculationData.price;
-            this.calculation.push(calculationData);
+        if (this.isCosmetic() && nonCosmetic) {
+            return null;
         }
-        const calculationData = {
-            id: 'HOT_POTATO_BOOK',
-            type: 'hot_potato_book',
-            price: (prices['hot_potato_book'] || 0) * Math.min(hotPotatoCount, 10) * applicationWorth.hotPotatoBook,
-            count: Math.min(hotPotatoCount, 10),
-        };
-        this.price += calculationData.price;
-        this.calculation.push(calculationData);
-    }
 
-    #hasRecomb() {
-        return this.extraAttributes.rarity_upgrades > 0 && !this.extraAttributes.item_tier;
-    }
+        this.#getBasePrice(prices, nonCosmetic);
 
-    #calculateRecomb(prices) {
-        const lastLoreLine = item.tag.display?.Lore?.at(-1);
-        if (
-            ExtraAttributes.enchantments ||
-            allowedRecombTypes.includes(skyblockItem?.category) ||
-            allowedRecombIds.includes(itemId) ||
-            lastLoreLine?.includes('ACCESSORY') ||
-            lastLoreLine?.includes('HATCESSORY')
-        ) {
-            const recombApplicationWorth = itemId === 'bone_boomerang' ? applicationWorth.recomb * 0.5 : applicationWorth.recomb;
-            const calculationData = {
-                id: 'RECOMBOBULATOR_3000',
-                type: 'recombobulator_3000',
-                price: (prices['recombobulator_3000'] || 0) * recombApplicationWorth,
-                count: 1,
-            };
-            this.price += calculationData.price;
-            this.calculation.push(calculationData);
+        const calculations = [HotPotatoBookCalculation, PickonimbusCalculation, Recombobulator3000Calculation];
+        for (const Calculation of calculations) {
+            Calculation.calculate(this);
```

Comment:
I like this as a static method call like is here 👍 

Would also need to pass prices ofc.